clambcc/clambc-compiler.py

```diff
@@ -46,6 +65,45 @@
 
 VERBOSE=False
 
+#Set of build tools needed on the system.
+class ClangLLVM():
+    def getLinkFromOpt(self, opt):
+        dirName = os.path.dirname(opt)
+        baseName = re.sub("opt", "llvm-link", os.path.basename(opt))
+        self.link = os.path.join(dirName, baseName)
+
+    def __init__(self, clangBinary, optBinary):
+        self.clang = clangBinary
+        self.opt = optBinary
+        self.link = None
+        self.getLinkFromOpt(self.opt)
+
+    def getClang(self):
+        return self.clang
+
+    def getOpt(self):
+        return self.opt
+
+    def getLLVMLink(self):
+        return self.link
+
+    def validate(self):
+        ov = findVersion(self.opt)
+        cv = findVersion(self.clang)
+        lv = findVersion(self.link)
+
+        if ov == -1:
+            print("error: unable to get version information for opt", file=sys.stderr)
+
+        if ov != cv:
```

Comment:
It may help to use an f-string here to print out which versions of each were found, in case it helps with debugging an issue in the future.